ethereum/contracts/EtherVault.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// Holds ETH on behalf of ETHApp
+contract EtherVault is Ownable {
+    event Deposit(address account, address sender, uint256 amount);
+    event Withdraw(address account, address recipient, uint256 amount);
+
+    receive() external payable {
+        revert("Must use deposit function");
+    }
+
+    // Accepts ETH from the caller.
+    function deposit(address _sender) 
+        external
+        payable
+        onlyOwner
+    {
+        emit Deposit(msg.sender, _sender, msg.value);
+    }
+
+    // Returns ETH to the caller.
+    function withdraw(address payable _recipient, uint256 _amount)
+        external
+        onlyOwner
+    {
+        require(_amount > 0, "Must unlock a positive amount");
+        (bool success, ) = _recipient.call{ value: _amount }("");
```

Comment:
Similar to [Ownable's `transferOwnership`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/access/Ownable.sol#L70), should we require the recipient to be non-zero?

My understanding is that the zero address is the burn address: nobody has the keys to it, so the funds can't be retrieved. It's unlikely that a user would want to transfer eth across the bridge only to immediately burn it. They can always burn the eth later themselves if that's really what they want.

If this is correct, we should also require a non-zero address in the [EthApp pallet's burn call](https://github.com/Snowfork/snowbridge/blob/main/parachain/pallets/eth-app/src/lib.rs#L122), to catch this before it gets across the bridge.

Reply:
None of our apps originally do this check. It may be useful but I think it is out of scope for this PR.